### PR TITLE
[NPU] fix accuracy npu op bug and change top_k's output to int64

### PR DESCRIPTION
**Pull request by pangyoki**
### PR types

Bug fixes

### PR changes

OPs

### Describe

- In `top_k` npu op, the dtype of the output `indices` is converted from int32 to int64 

图片

- fix bugs of `accuracy` npu op

单测结果：
图片

日志：

图片

### CODE REVIEW

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc [hunk 1]
```diff
@@ -23,91 +23,84 @@ template <typename DeviceContext, typename T>
...
+    FillNpuTensorWithConstant<int>(total, static_cast<int>(num_samples));
+
+    // cast total to float32 for calculating accuracy
+    Tensor tmp_total(framework::proto::VarType::FP32);
+    tmp_total.Resize(total->dims());
+    tmp_total.mutable_data<float>(ctx.GetPlace());
+    auto runner_cast_total = NpuOpRunner(
+        "Cast", {*total}, {tmp_total},
+        {{"dst_type", static_cast<int>(ConvertToNpuDtype(tmp_total.type()))}});
+    runner_cast_total.Run(stream);
```
Comment (zhiqiu) [inline paddle/fluid/operators/metrics/accuracy_op_npu.cc]: why not call FillNpuTensorWithConstant<float> directly?

Comment (pangyoki) [reply paddle/fluid/operators/metrics/accuracy_op_npu.cc]: changed `Cast` NPU OP to `FillNpuTensorWithConstant`

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc [hunk 2]
```diff
@@ -23,91 +23,83 @@ template <typename DeviceContext, typename T>
...
+
+    // [total]
+    total->mutable_data<int>(ctx.GetPlace());
+    FillNpuTensorWithConstant<int>(total, static_cast<int>(num_samples));
+
+    // use `total` of type `float32` for calculating accuracy
+    Tensor tmp_total(framework::proto::VarType::FP32);
+    tmp_total.Resize(total->dims());
+    tmp_total.mutable_data<float>(ctx.GetPlace());
+    tmp_total.mutable_data<float>(ctx.GetPlace());
```
Comment (zhiqiu) [inline paddle/fluid/operators/metrics/accuracy_op_npu.cc]: duplicate line

**Review by zhiqiu — APPROVED** [review]
LGTM

File: paddle/fluid/operators/metrics/accuracy_op_npu.cc (hunk 2 above)
Comment (pangyoki) [reply paddle/fluid/operators/metrics/accuracy_op_npu.cc]: done

**Pull request merged**